ibis/backends/flink/__init__.py

```diff
@@ -635,6 +638,126 @@ def drop_view(
         sql = statement.compile()
         self._exec_sql(sql)
 
+    def _read_file(
+        self,
+        file_type: str,
+        path: str | Path,
+        schema: sch.Schema,
+        table_name: str | None = None,
+    ) -> ir.Table:
+        """Register a file as a table in the current database.
+
+        Parameters
+        ----------
+        file_type
+            File type, e.g., parquet, csv, json.
+        path
+            The data source.
+        schema
+            The schema for the new table.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        table_name = table_name or gen_name(f"read_{file_type}")
+        tbl_properties = {
+            "connector": "filesystem",
+            "path": path,
+            "format": file_type,
+        }
+
+        return self.create_table(
+            name=table_name,
+            schema=schema,
+            tbl_properties=tbl_properties,
+        )
+
+    def read_parquet(
+        self,
+        path: str | Path,
+        schema: sch.Schema,
+        table_name: str | None = None,
+    ) -> ir.Table:
+        """Register a parquet file as a table in the current database.
+
+        Parameters
+        ----------
+        path
+            The data source.
+        schema
+            The schema for the new table.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        return self._read_file(
+            file_type="parquet", path=path, schema=schema, table_name=table_name
+        )
+
+    def read_csv(
+        self,
+        path: str | Path,
+        schema: sch.Schema,
+        table_name: str | None = None,
+    ) -> ir.Table:
+        """Register a csv file as a table in the current database.
+
+        Parameters
+        ----------
+        path
+            The data source.
+        schema
+            The schema for the new table.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        return self._read_file(
+            file_type="csv", path=path, schema=schema, table_name=table_name
+        )
+
+    def read_json(
+        self,
+        path: str | Path,
+        schema: sch.Schema,
```

Comment:
For all of these can you mark the `schema` parameter as

```suggestion
        schema: sch.Schema | None = None,
```

And then error nicely inside the `_read_file` method saying that the `schema` parameter is required for the `flink` backend? This keeps the method/function signatures the same across backends and provides a nicer error to the end user.

Reply:
Thanks, I made the requested change.